### PR TITLE
Rewrite the README; Rename to WebAuthn; Proper Copyright

### PR DESCRIPTION
**Pull request by ErinvanderVeen**
This MR contains several changes as listed:
- Rewrite the README: completely rewrote the readme to include the binary caches
- Rename the project to WebAuthn, we discussed that Fido2 is not the right name for the library. This does clash with the existing `webauthn` libary, I propose discussing this with the maintainer of the existing library.
- We add a copyright statement, attributing copyright from 2020 to 2021 to @arianvp , the original author of the library, and 2021 and onwards to Tweag I/O

### CODE REVIEW

**Comment by Infinisil** [comment]
I think the readme structure would be good like this:
# Haskell Webauthn Library
## Development environment
### Nix caches
### Nix shell
## Developing the library
## Running the tests
## Running the demo
## License

File: README.md [hunk 1]
```diff
@@ -1,63 +1,98 @@
...
+instructions below assume the use of Nix.
+
+### Nix Caches
+We highly recommend using the IOHK Binary Cache to avoid building of several
+copies of GHC. Setting up the IOHK binary cache is easily done following the
+[instructions][cache] provided by IOHK.
+
+Additionally, Tweag provides a Cachix cache for the library itself and all
+non-Haskell dependencies, setting up the `tweag-haskell-fido2` cache can be
+done using the [instructions][cachix] provided by Cachix.
```
Comment (Infinisil) [inline README.md:21]: You can link to https://app.cachix.org/cache/tweag-haskell-fido2 directly

File: README.md [hunk 2]
```diff
@@ -1,63 +1,98 @@
...
+
+## Developing the Library
+The [Haskell Language Server (hls)][hls] and [Ormolu][ormolu] are highly recommended
+for the development of this library. To configure hls we would refer you to the
+instructions specific to your editor. We also recommend enabling auto-format
+using Ormolu for your editor. We do, however, also provide a bash script in
+`bin/autoformat.sh` that uses Ormolu to format all Haskell source files in the
+repository.
+
+Code not formatted using Ormolu is likely to be rejected in CI.
```
Comment (Infinisil) [inline README.md:40]:
```suggestion
Code not formatted using Ormolu will be rejected in CI.
```

**Review by Infinisil — CHANGES_REQUESTED** [review]

**Comment by arianvp** [comment]
Hmm it's a shame that we ended up with two implementations of webauthn in parallel.  Is there an intention to merge the https://hackage.haskell.org/package/webauthn project and this one?

**Comment by arianvp** [comment]
Ah it actually seems to be quite a bit older but just got a lot of recent development. I didn't spot it when I started this library

**Comment by Infinisil** [comment]
Oh yeah, we were actually in contact with @fumieval from https://github.com/fumieval/webauthn as well, indicating that we'd be working on an updated implementation. So I'm surprised that they starting working on their implementation as well without mentioning it to us. I'd love to unify all of https://github.com/arianvp/haskell-fido2, https://github.com/tweag/haskell-fido2 and https://github.com/fumieval/webauthn into a single library, to be put in https://hackage.haskell.org/package/webauthn.

@fumieval Perhaps we haven't informed you well enough of our efforts, but we've been working on this library here for some months now and we think it's near production-readiness! What are your thoughts?

**Comment by lykahb** [comment]
A name that contains "WebAuthn" is more accurate because this library does not deal with CTAP, the aspect of FIDO2 about connecting client and authenticator. It may also make it easier to discover the library on Hackage. I don't have an opinion about it being exactly "webauthn" or something like "webauthn-suffix".

Also, it is more straightforward to have the same name for the project and the git repository. Once a new name is confirmed, it's worth renaming a repository too. When I first started using this library, I added a bad dependency haskell-fido2 instead of fido2.

**Comment by ErinvanderVeen** [comment]
@fumieval We are still working on adding TPM and AndroidSafetyNet attestation. It is nice to hear that we can use the webauthn name for the package, thank you. We will contact you again whenever we reach that point.

**Comment by fumieval** [comment]
@ErinvanderVeen Great. Once the time is ripe, let me know your Hackage user names and I can add you to the uploaders at any point

**Comment by Infinisil** [comment]
@fumieval Awesome thanks! Also feel free to let us know if you see anything in our implementation that you'd like to see changed/improved.

**Pull request merged**